### PR TITLE
React/week2

### PR DESCRIPTION
**Pull request by ember-b-Moss**

### CODE REVIEW

File: react/week2/to-do/src/components/List.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React, { useState } from "react";
+
+function List(props) {
+  const [state, setDone] = useState([...props.items]);
```
Comment (piyushpalod) [inline react/week2/to-do/src/components/List.js:4]: may be better variable name here will help in readability.

File: react/week2/to-do/src/components/List.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      if (item.id === id) {
+        return { ...item, done: !item.done };
+      }
+      return item;
+    });
+    setDone(newState);
+  };
+
+  const deleteItem = (id) => {
+    setDone(state.filter((item) => item.id !== id));
```
Comment (piyushpalod) [inline react/week2/to-do/src/components/List.js:42]: Can you please explain this , to delete Item you need to remove it from your array. how you are removing from here ?

File: react/week2/to-do/src/components/ListSection.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import List from "./List";
+
+function ListSection(props) {
+  return (
+    <section className="List">
```
Comment (piyushpalod) [inline react/week2/to-do/src/components/ListSection.js:6]: where is section component ?

**Review by piyushpalod — COMMENTED** [review]
I have few comment, can you please check and reply. otherwise PR looks good.

File: react/week2/to-do/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+function Header() {
+  return (
+    <header className="App-header">
```
[the remark this answers is not in the document]
Comment (ember-b-Moss) [reply react/week2/to-do/src/components/Header.js:5]: just gave it a class so i can style it later

File: react/week2/to-do/src/components/List.js (hunk 2 above)
Comment (ember-b-Moss) [reply react/week2/to-do/src/components/List.js:42]: by id. it find the item with the id connected to item that user wants to delete. i filter the array with items and the one with requested id gets out. i mean a new array forms with all the items with ids that dont match this id

File: react/week2/to-do/src/components/ListSection.js (hunk 3 above)
Comment (ember-b-Moss) [reply react/week2/to-do/src/components/ListSection.js:6]: in a separate js. file

**Pull request merged**